packages/core/src/intl/languages.ts

```diff
@@ -102,14 +102,26 @@ export function getLanguageByShortRegionCode<
     Language extends Pick<NewsroomLanguageSettings, 'is_default' | 'code' | 'public_stories_count'>,
 >(languages: Language[], locale: LocaleObject): Language | undefined {
     const shortRegionCode = locale.toRegionCode();
+    const neutralLanguageCode = locale.toNeutralLanguageCode();
 
-    // Prefer default language
+    // 1. Prefer language with same neutral language code
```

Comment:
This looks incorrect. The method is specifically designed to return a language with matching REGION code part.

To me, the issue is in the order in which this function is executed.
It should be as simple as switching the order of functions in [getLanguageFromNextLocaleIsoCode](https://github.com/prezly/theme-kit-js/blob/c6fff3d81acac45c469248566e9e04625c125157/packages/nextjs/src/intl/languages.ts#L32) function.

However, I can see that the current order of checks in that function corresponds to `getShortestLocaleCode` checks.

All in all, I feel like this requirement of shorting to region code is a huge pain in the butt, and we should get rid of this logic. This used to be a request from older customers, not sure how critical is this nowadays.

Reply:
As a middle-ground solution, I think you can extract this logic into a separate method and call that one in `getLanguageFromNextLocaleIsoCode` right after exact locale code check.
I'm afraid that hacks like this would make things even worse in the long term

Reply:
> This looks incorrect. The method is specifically designed to return a language with matching REGION code part.

Yeah, I also thought about this. But what about the other solution (#297)?

That's exactly what I did there.